File: CbeMeetingAttendance/Model/Employee.cs
using System.ComponentModel.DataAnnotations;

namespace CbeMeetingAttendance.Models
{
    public class Employee
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        [MaxLength(50)]
        public string EmployeeId { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        public string FullName { get; set; } = string.Empty;

        [MaxLength(100)]
        public string Division { get; set; } = string.Empty;

        [MaxLength(100)]
        public string JobTitle { get; set; } = string.Empty;

        [MaxLength(100)]
        public string Department { get; set; } = string.Empty;

        [MaxLength(20)]
        public string PhoneNumber { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation Property
        public ICollection<Attendance> Attendances { get; set; } = new List<Attendance>();
    }
}

File: mobileapp/lib/models/employee.dart
/// Backend EmployeeResponseDto.
class Employee {
  Employee({
    required this.id,
    required this.employeeId,
    required this.fullName,
    this.division = '',
    this.jobTitle = '',
    this.department = '',
    this.phoneNumber = '',
  });

  final String id; // GUID
  final String employeeId; // e.g. EMP001
  final String fullName;
  final String division;
  final String jobTitle;
  final String department;
  final String phoneNumber;

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      id: json['id']?.toString() ?? '',
      employeeId: json['employeeId'] as String? ?? '',
      fullName: json['fullName'] as String? ?? '',
      division: json['division'] as String? ?? '',
      jobTitle: json['jobTitle'] as String? ?? '',
      department: json['department'] as String? ?? '',
      phoneNumber: json['phoneNumber'] as String? ?? '',
    );
  }
}
